### DISCUSSION TITLE
more scoped features

### DISCUSSION THREAD

**Discussion by danielo515** [johnlindquist/kit#1123] [Ideas]
Hello.
Maybe the intention of scriptkit is not to replace something like raycast, in which case you can just say that and this discussion will probably be over.
But, if you are open to it, I think there are some raycast ideas that may be good to be added to scriptkit.
One of those things is the existence of different kinds of "elements" within the app.
For example, raycast has quick-links, that are just direct access to certain elements, either be web urls, files or programs. They are handy and they are really easy to create. Of course I could create a script for each one of those in scriptkit, but I think it defeats the convenience factor.
A script to handle those can also be created, but then you will not be able to find them directly, you will have to first open the script, and then find your quicklink.
Maybe all is needed is one API for scripts to register subcommands, just like raycast extensions do.

**Comment by Ambushfall** [comment]
This sounds like a pretty cool idea, could be represented as a different tab in scriptkit as well.

**Comment by johnlindquist** [comment]
@danielo515 @Ambushfall 

The original idea of Script Kit was to allow enough flexibility that people could build their own "Main Prompt":

```js
import "@johnlindquist/kit"

let scripts = await getScripts()

let links = [`https://scriptkit.com`, `https://egghead.io`]

let choices = [...links, ...scripts]

let choice = await arg("Run", choices)

if (typeof choice === "string") {
  open(choice)
} else {
  await run(choice.filePath)
}
```

You then assign a `// Shortcut` and you've got your own menu.

## The Main Prompt is "Just a Script"

If you look at `~/.kit/main/index.js`, you'll see the script that makes up the "Main" prompt. People can tear it apart and rebuild it however they want.

## Configuration vs. Code

I'm heavily in the camp of "code over configuration", but I'm always open to user feedback. It would be fairly easy to add a `~/.kenv/links.json` that allows people to define links that will be auto-loaded into the main menu. We could even add a "manage links" from the Kit tab or something that easily allows you to add/remove the links. Lots of possibilities.

## Root Search/Mega Menu

I waver back and forth between allowing searching for _everything_ (apps, links, commands, etc) in the main menu. I know that's how Raycast/Alfred/etc approach it, but I've always been in the camp of pressing a single key to switch search context (`;` for apps, `/` for directories, etc). I know people expect Script Kit to behave like Raycast, etc, but I try to avoid comparing to Raycast, etc because Script Kit is meant to focus on being a "Kit for Scripts" not "another launcher". I know that line gets blurrier every day
